user/test.py

```diff
@@ -0,0 +1,55 @@
+import json
+from unittest import TestCase
+
+from core import app
+from blog import views
+from user.models import User
+from trips import views
+from user import views
+from auth import views
+
+
+def login_user(self):
+    data = {
+        'email': 'emailtest@test.com',
+        'password': 'testtest',
+    }
+    self.client.post('/login', data=data, follow_redirects=True)
+
+
+class UserTest(TestCase):
+    def setUp(self):
+        self.client = app.test_client()
+        self.user = User(email='emailtest@test.com').set_password('testtest')
+        self.user.allow_localization = True
+        self.user.save()
+        self.lat_lng = [3.5, 42.0]
+
+    def tearDown(self):
+        User.objects.delete()
+
+    def test_user_page_that_does_not_exist(self):
+        result = self.client.get('/@doesnotexist')
+        self.assertEqual(result.status_code, 404)
+
+    def test_privacy_page_unauthorized(self):
+        result = self.client.get('/privacy')
+        self.assertEqual(result.status_code, 401)
+
+    def test_privacy_page(self):
+        login_user(self)
+        result = self.client.get('/privacy')
+        self.assertEqual(result.status_code, 200)
+
+    def test_delete_trip(self):
+        login_user(self)
+        lat_lng = [10, 10]
+        self.client.post('/trips/add', data=json.dumps(lat_lng),
+                         content_type='application/json')
+        user = User.objects.first()
+        url = '/privacy/[10, 10]/' + user.locations[0]\
```

Comment:
Ideally I would use the timestamp that is more url-friendly and unique enough:
```python
url = '/privacy/{}'.format(user.location[0].date.timestamp())
```

However, if for some reason you could not do that, at least formatting the url a little cleaner:

```python
url = '/privacy/10,10/{}'.format(user.location[0].date.strftime('%Y-%m-%d %H:%M:%S.%f'))
```

But really the first way may be more appropriate